File: Core/GamepadButton.swift
/// Every physical button on the 8BitDo Micro that the app can respond to.
///
/// Using a dedicated enum rather than the raw GameController framework types
/// keeps mode definitions clean and decoupled from Apple's framework internals.
/// If the controller hardware changes, only `ControllerManager` needs updating.
enum GamepadButton: String, CaseIterable, Hashable {

    // MARK: - Face Buttons

    case a = "A"
    case b = "B"
    case x = "X"
    case y = "Y"

    // MARK: - D-Pad

    case dpadUp    = "D-Pad Up"
    case dpadDown  = "D-Pad Down"
    case dpadLeft  = "D-Pad Left"
    case dpadRight = "D-Pad Right"

    // MARK: - Shoulder Buttons

    case leftShoulder  = "L1"
    case rightShoulder = "R1"
    case leftTrigger   = "L2"
    case rightTrigger  = "R2"

    // MARK: - Menu Buttons

    case start  = "Start"
    case select = "Select"

    // MARK: - Computed Properties

    /// A display-friendly label for preferences UI and accessibility labels.
    var displayName: String { rawValue }
}

File: pocket_study_remote/core/gamepad_button.py
"""Typed enum for every physical button on the 8BitDo Micro."""

from enum import Enum


class GamepadButton(Enum):
    """
    Every button the app can respond to.

    Values are display-friendly strings used in logs and (eventually)
    the preferences UI.  The actual mapping from hardware indices to
    these enum members lives in ControllerManager — nowhere else.
    """

    # Face buttons
    A = "A"
    B = "B"
    X = "X"
    Y = "Y"

    # D-pad
    DPAD_UP    = "D-Pad Up"
    DPAD_DOWN  = "D-Pad Down"
    DPAD_LEFT  = "D-Pad Left"
    DPAD_RIGHT = "D-Pad Right"

    # Shoulder buttons
    LEFT_SHOULDER  = "L1"
    RIGHT_SHOULDER = "R1"
    LEFT_TRIGGER   = "L2"
    RIGHT_TRIGGER  = "R2"

    # Menu buttons
    START  = "Start"
    SELECT = "Select"

    @property
    def display_name(self) -> str:
        return self.value
